app/code/Magento/Customer/Model/Indexer/Source.php

```diff
@@ -61,6 +61,22 @@ public function addFieldToSelect($fieldName, $alias = null)
         return $this;
     }
 
+    /**
+     * Add attribute to entities in collection
+     *
+     * If $attribute == '*' select all attributes
+     *
+     * @param array|string|integer|\Magento\Framework\App\Config\Element $attribute
+     * @param bool|string $joinType flag for joining attribute
+     * @return $this
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function addAttributeToSelect($attribute, $joinType = false)
```

Comment:
The fix provided by support is a workaround and should not be applied in core. `Magento\Framework\Indexer\Handler\AttributeHandler::prepareSql` uses `Magento\Framework\App\ResourceConnection\SourceProviderInterface` so can not know about method `\Magento\Customer\Model\Indexer\Source::addAttributeToSelect`. To fix an issue use `Magento\Customer\Model\Indexer\Source::addFieldToSelect` in `Magento\Framework\Indexer\Handler\AttributeHandler::prepareSql`